File: SportsClub.Domain/Entities/Operations/Shuttle.cs
namespace SportsClub.Domain.Entities.Operations;

public class Shuttle : BaseEntity
{
    public string LicensePlate { get; set; } = string.Empty;
    public string DriverName { get; set; } = string.Empty;
    public string DriverPhone { get; set; } = string.Empty;
    public string RouteName { get; set; } = string.Empty; // Örn: Kadıköy-Bostancı Güzergahı
    public int Capacity { get; set; }
    
    // Servise kayıtlı öğrenciler
    public ICollection<ShuttleStudent> ShuttleStudents { get; set; } = new List<ShuttleStudent>();
}

File: SportsClub.Frontend/src/types/shuttle.ts
export interface Shuttle {
    id: string;
    licensePlate: string;
    driverName: string;
    driverPhone: string;
    routeName: string;
    capacity: number;
    enrolledStudentCount: number;
}

export interface CreateShuttleDto {
    licensePlate: string;
    driverName: string;
    driverPhone: string;
    routeName: string;
    capacity: number;
}

export interface EnrollStudentDto {
    athleteId: string;
    pickUpDropOffPoint: string;
}
